playgame.py

```diff
@@ -24,6 +25,11 @@ def main(argv):
     else:
         run_one_game(args)
 
+def ensure_path(path):
+    directories_from_path = os.path.dirname(path)
+    if directories_from_path and not os.path.isdir(directories_from_path):
```

Comment:
got a little hands-on. I was working off a bad assumption and I think that was leading to some confusion

You do need to check that directories_from_path is not empty, but the pythonic way is just to check it directly. This is the clean way to write this line. We should be doing the empty check prior to the isdir check to short-circuit. I think the failing function is makedirs, but its more readable anyways